main.go

```diff
@@ -16,13 +16,22 @@ import (
 )
 
 func Main() error {
+	return MainWithOptions(&Options{})
+}
+
+type Options struct {
+	Transport http.RoundTripper
+}
+
+func MainWithOptions(options *Options) error {
```

Comment:
I prefer to add the argument directly to the `Main` function.

```suggestion
type Options struct {
	Transport http.RoundTripper
}

func Main(options *Options) error {
```

Reply:
Please write a comment about this argument, because later when we read the code, we may wonder what the argument of the Main function is used for.

Reply:
Done.